MoEngagePluginBase/MoEngagePluginUtils.swift

```diff
@@ -0,0 +1,161 @@
+//
+//  MoEngagePluginUtils.swift
+//  MoEngagePlugin
+//
+//  Created by Rakshitha on 23/06/22.
+//
+
+import Foundation
+import MoEngageInApps
+import MoEngageSDK
+
+public protocol MoEngagePluginUtils {
+    static func fetchIdentifier(attribute: [String: Any]) -> String?
+    static func fetchAccountPayload(identifier: String) -> [String: Any]
+    static func fetchInAppPayload(inAppCampaign: MOInAppCampaign, inAppAction: MOInAppAction?, identifier: String) -> [String: Any]
+    static func fetchSelfHandledPayload(selfHandledCampaign: MOInAppSelfHandledCampaign?, identifier: String) -> [String: Any]
+    static func fetchTokenPayload(deviceToken: String) -> [String: Any]
+    static func fetchPushClickedPayload(withScreenName screenName: String?, kvPairs: [AnyHashable: Any]?, andPushPayload userInfo: [AnyHashable: Any], identifier: String) -> [String: Any]
+}
+
+extension MoEngagePluginUtils {
+    
+    // MARK: General Utilities
+    public static func fetchIdentifier(attribute: [String: Any]) -> String? {
```

Comment:
could be named `getIdentifierFromPayload()` fetch gives an impression that the value is being read from some storage or remote server.

Reply:
resolved